python/ujson.c

```diff
@@ -46,19 +46,23 @@ void initObjToJSON(void);
 /* objToJSONFile */
 PyObject* objToJSONFile(PyObject* self, PyObject *args, PyObject *kwargs);
 
-HPy_DECL_METH_O(JSONToObj)
-HPy_DECL_METH_O(JSONFileToObj)
-
+extern HPyDef JSONToObj;
+extern HPyDef JSONToObj_decode;
+extern HPyDef JSONFileToObj;
+
+static HPyDef *module_defines[] = {
+    &JSONToObj,
+    &JSONToObj_decode,
+    &JSONFileToObj,
+    NULL
+};
 
 #define ENCODER_HELP_TEXT "Use ensure_ascii=false to output UTF-8. Pass in double_precision to alter the maximum digit precision of doubles. Set encode_html_chars=True to encode < > & as unicode escape sequences. Set escape_forward_slashes=False to prevent escaping / characters."
 
-static HPyMethodDef ujsonMethods[] = {
-  {"encode", (HPyMeth) objToJSON, METH_VARARGS | METH_KEYWORDS, "Converts arbitrary object recursively into JSON. " ENCODER_HELP_TEXT},
-  {"decode", JSONToObj, HPy_METH_O, "Converts JSON as string to dict object structure. Use precise_float=True to use high precision float decoder."},
-  {"dumps", (HPyMeth) objToJSON, METH_VARARGS | METH_KEYWORDS,  "Converts arbitrary object recursively into JSON. " ENCODER_HELP_TEXT},
-  {"loads", JSONToObj, HPy_METH_O,  "Converts JSON as string to dict object structure. Use precise_float=True to use high precision float decoder."},
-  {"dump", (HPyMeth) objToJSONFile, METH_VARARGS | METH_KEYWORDS, "Converts arbitrary object recursively into JSON file. " ENCODER_HELP_TEXT},
-  {"load", JSONFileToObj, HPy_METH_O, "Converts JSON as file to dict object structure. Use precise_float=True to use high precision float decoder."},
+static PyMethodDef ujsonMethods[] = {
+  {"encode", (PyCFunction) objToJSON, METH_VARARGS | METH_KEYWORDS, "Converts arbitrary object recursively into JSON. " ENCODER_HELP_TEXT},
```

Comment:
Antonio already ported the _encode_ part in branch https://github.com/hpyproject/ultrajson-hpy/tree/feature/convert-encode-to-hpy . I think we should merge the branch.